src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -746,6 +746,10 @@ public static function export($value): string
             return 'null';
         }
         if (!\is_array($value)) {
+            if (\is_object($value)) {
```

Comment:
`if (\is_object($value) && !method_exists($value, '__set_state')) {`

Reply:
even objects with `__set_state` are broken, as the logic currently replaces new lines by a concatenation with `"\n"` instead (because when var_export-ing a scalar, new lines appear only inside strings).

If we want to keep support for objects with `__set_state`, it should be handled as a separate case (and I would put it before the `if (!is_array())` case then, rather than nesting in it)

Reply:
> even objects with __set_state are broken

indeed, then let's keep it unsupported :)

Reply:
The purpose of this PR is to disallow object usage, and avoid to export objects (via `var_export`). Thus, the fact that `__set_state` exists or not is not relevant. The dumped file does not work when we export an object (because of `\n` replacements).